endpoints/version.go

```diff
@@ -7,31 +7,33 @@ import (
 	"github.com/golang/glog"
 )
 
-type versionModel struct {
-	Version  string `json:"version"`
-	Revision string `json:"revision"`
-}
+const versionNotSet = "not-set"
 
 // NewVersionEndpoint returns the latest git tag as the version and commit hash as the revision from which the binary was built
-func NewVersionEndpoint(version string, revision string) http.HandlerFunc {
-	if version == "" {
-		version = "not-set"
-	}
-	if revision == "" {
-		revision = "not-set"
+func NewVersionEndpoint(version, revision string) http.HandlerFunc {
+	response, err := prepareVersionResponse(version, revision)
+	if err != nil {
+		glog.Fatalf("error creating /version endpoint response: %v", err)
 	}
 
 	return func(w http.ResponseWriter, _ *http.Request) {
-		jsonOutput, err := json.Marshal(versionModel{
-			Version:  version,
-			Revision: revision,
-		})
-		if err != nil {
-			glog.Errorf("/version Critical error when trying to marshal versionModel: %v", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
+		w.Write(response)
+	}
+}
 
-		w.Write(jsonOutput)
+func prepareVersionResponse(version, revision string) (json.RawMessage, error) {
+	if version == "" {
+		version = versionNotSet
 	}
+	if revision == "" {
+		revision = versionNotSet
```

Comment:
Nitpick: This could be a bit confusing. Ideally we should assign `revisionNotSet` and not `versionNotSet` to `revision`. Do you think renaming `versionNotSet` to either `notSet` or `valueNotSet` could be more straightforward?

Reply:
Renamed for clarity. I cannot rename it `notSet` or `valueNotSet` because it's in the same package (naming scope) as other endpoints.